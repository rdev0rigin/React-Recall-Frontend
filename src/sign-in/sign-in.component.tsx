import * as React from 'react';
import { RecallStore } from '../services/store.redux';
import {loadGapi} from '../services/gapi.service';
// import { actionSignIn } from '../main/app.actions';
// import { loadGapi } from '../services/gapi.service';
// import { transformGoogleDataToUserModel } from '../models/user.model';
// import { responseSocket } from '../services/socket-io.service';
declare var gapi: any;

export class SignInComponent extends React.Component {

	public state: {
		gapiReady: boolean;
	};

	constructor(
		public props: any
	) {
		super(props);
		this.signInUpHandler = this.signInUpHandler.bind(this);
		this.state = RecallStore.getState();
	}

	public componentDidMount(): void {
		if(!this.state.gapiReady){
			loadGapi().subscribe(loaded => {
				if(loaded) {
					this.setState(RecallStore.getState());
					this.props.history.push('/');
				}
			});
		}
	}

	public signInUpHandler(e: any): void {
		e.preventDefault();
		if(this.state.gapiReady){
			gapi.auth2
				.getAuthInstance()
				.signIn({
					prompt: 'consent',
					ux_mode: 'popup'
				})
				.then((profileInfo: any) => {
					console.log('response?');
					console.log('current user?', gapi.auth2.getAuthInstance().currentUser.get());
					console.log('profile?', profileInfo);
				})
				.catch((res: any) => console.log('auth error', res));
		}
	}

	render(){
		return(
			<div className={'sign-in-container'}>
				{this.state.gapiReady ? 'Ready!' : 'Not Ready'}
				<h1>Sign In</h1>
				<form>
					<button onClick={this.signInUpHandler}>Sign In With Google</button>
				</form>
			</div>
		)
	}
}