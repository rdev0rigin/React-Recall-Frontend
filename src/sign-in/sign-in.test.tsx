import * as React from 'react';
import {mount, shallow} from 'enzyme';
import {SignInComponent} from './sign-in.component';

describe('Sign In Component Tests', () => {
	it('should render', () => {
		const component = shallow(<SignInComponent/>);
		expect(component).toMatchSnapshot();
	});

	it('should initialize with state, gapiReady to be false', () => {
		const component = shallow(<SignInComponent/>);
		expect(component.state().gapiReady).toBe(false);
	});

	it('should call loadGapi() on componentDidMount', () => {
		const component = mount(<SignInComponent/>);
		console.log(component.props);
	})

});