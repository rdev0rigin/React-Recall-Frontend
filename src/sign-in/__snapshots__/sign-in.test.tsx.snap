// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign In Component Tests should render 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="sign-in-container"
>
    Not Ready
    <h1>
        Sign In
    </h1>
    <form>
        <button
            onClick={[Function]}
        >
            Sign In With Google
        </button>
    </form>
</div>,
  "nodes": Array [
    <div
      className="sign-in-container"
>
      Not Ready
      <h1>
            Sign In
      </h1>
      <form>
            <button
                  onClick={[Function]}
            >
                  Sign In With Google
            </button>
      </form>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <SignInComponent />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": SignInComponent {
        "_reactInternalInstance": [Circular],
        "assignGoogleAuth2": [Function],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "gapiReady": false,
        },
        "updateSigninStatus": [Function],
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="sign-in-container"
>
          Not Ready
          <h1>
                    Sign In
          </h1>
          <form>
                    <button
                              onClick={[Function]}
                    >
                              Sign In With Google
                    </button>
          </form>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="sign-in-container"
>
          Not Ready
          <h1>
                    Sign In
          </h1>
          <form>
                    <button
                              onClick={[Function]}
                    >
                              Sign In With Google
                    </button>
          </form>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <SignInComponent />,
}
`;
